docs/src/pages/components/data-grid/rendering/DensitySelectorGrid.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import {
+  DataGrid,
+  GridToolbarContainer,
+  DensitySelector,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+
+export default function DensitySelectorGrid() {
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 4,
+    maxColumns: 6,
+  });
+
+  return (
+    <div style={{ height: 300, width: '100%' }}>
+      <DataGrid
+        {...data}
+        components={{
+          Toolbar: () => (
+            <GridToolbarContainer>
+              <DensitySelector />
+            </GridToolbarContainer>
+          ),
```

Comment:
We need to create a new component. We can't unmount at each render.

Reply:
Done